lib/util/experimental-flags.ts

```diff
@@ -0,0 +1,17 @@
+import { GlobalConfig } from '../config/global';
+
+export function experimentalFlagValue(flagName: string): string | null {
+  const experimentalFlags = GlobalConfig.get('experimentalFlags');
+  if (!experimentalFlags) {
+    return null;
+  }
+
+  for (const flag of experimentalFlags) {
+    if (flag.includes(flagName)) {
+      const [key, value] = flag.split('=');
+      return value ? value : key;
+    }
+  }
```

Comment:
can we do the parsing once? I see a performance issue when we don't on every call. 🤔